docs/github-merge-strategy.md

```diff
@@ -6,6 +6,43 @@ This document describes the merge strategy used for Pull Requests within all rep
 
 Every Pull Request that is made to an InstructLab repository should meet the below requirements - certain repositories such as [taxonomy](https://github.com/instruct-lab/taxonomy) may have additional requirements.
 
+### Commit Messages and PR Description
+
+PRs should be formatted as a series of logical commits that implement the
+overall change proposed in the PR. Each commit should represent a logical step
+in the progression of the change. The commit message for each should clearly
+explain that step of the progression.
```

Comment:
Do you want to be explicit here about squashing fixes and cleanups into those logical commits?

I look for high-level context about "what" a change is to frame my review, as well as some reason "why" the change is needed to understand prioritization and benefit.

Reply:
Good idea.  I have some older text at the bottom of this doc that could be moved up here as a start.  I already moved up some of it.  Here's what could be moved:

> It is common that a PR author may need to do a final rebase to clean up their proposed commit series before a PR can be merged. It is also fine for a project maintainer to perform this step when the changes necessary are straight forward enough to do so.  This includes doing a final rebase on `main` if necessary. The PR itself should NOT include any merge commits of `main` back into the developer's branch. We expect the proposed commit series to be a clean set of commits against `main` without conflicts or merge commit history. We only use a merge commit to record the PR's inclusion into `main`.